debtap: add page (#7947)
tldr-pages/tldr: pages/linux/debtap.md

# debtap

> Convert Debian packages into Arch Linux packages.
> See also: `pacman-upgrade`.
> More information: <https://github.com/helixarch/debtap>.

> - Update debtap database (before first run):

Reviewer: ```suggestion
- Update debtap database (before the first run):
```

I think it sounds better.

`sudo debtap --update`

- Convert the specified package:

`debtap {{path/to/package.deb}}`

> - Convert the specified package (bypass all questions, not recommended):

Reviewer 2: ```suggestion
- Convert the specified package bypassing all questions, except for editing metadata files:
```

Author: Following the contributing guidelines, I think it's better to keep it short and simple 😃

Author: For convenience, just skip them all and install to the local machine later on.

# debtap

> > Convert .deb packages into Arch Linux packages.

Reviewer 2: ```suggestion
> Convert Debian packages into Arch Linux packages.
```

> > More information: <https://github.com/helixarch/debtap>.

Reviewer 2: ```suggestion
> See also: `pacman-upgrade`.
> More information: <https://github.com/helixarch/debtap>.
```

Reviewer: Maybe better to suggest several `pacman*` commands at once.

Reviewer 2: But if you use this command you want to install the package, so `pacman-upgrade` is the direct access to it. Also as a user, you already know about `pacman` because you're in Arch Linux. Then, the `pacman-upgrade` page sounds as the best recommendation.

- Update debtap database (before first run)

> `sudo debtap -u`

Reviewer 2: We prefer long arguments if available
```suggestion
`sudo debtap --update`
```

> `sudo debtap -u`

Reviewer: ```suggestion
`sudo debtap --update`
```

We prefer long options.

- Convert a package

> `debtap {{package.deb}}`

Reviewer: ```suggestion
- Convert the specified package:

`debtap {{path/to/package.deb}}`
```

> `debtap {{package.deb}}`

Reviewer 2: ```suggestion
`debtap {{path/to/package.deb}}`
```

- Convert a package (bypass all questions, not recommended)

> `debtap -Q {{package.deb}}`

Reviewer 2: We should recommend best practices, just `--quiet` sounds better
```suggestion
`debtap --quiet {{path/to/package.deb}}`
```

> `debtap -Q {{package.deb}}`

Reviewer: ```suggestion
- Convert the specified package (bypass all questions, not recommended):

`debtap --quiet {{path/to/package.deb}}`
```

- Then install with pacman

> `sudo pacman -U {{package.pkg.tar.zst}}`

Reviewer: ```suggestion
- Install the specified package with `pacman`:

`sudo pacman -U {{package.pkg.tar.zst}}`
```

> `sudo pacman -U {{package.pkg.tar.zst}}`

Reviewer 2: We don't use to add examples to other commands, that's why we can add the `See also` note. Please remove this example.

- Generate PKGBUILD file

> `debtap -P {{package.deb}}`

Reviewer 2: We use `path/to/file` as placeholder
```suggestion
`debtap --pkgbuild {{path/to/package.deb}}`
```

Reviewer: We added the same suggestions. 😄

Reviewer 2: Reviewers attacking a first-time contributor. :heart:

Author: Than you both for your advice 😂

> `debtap -P {{package.deb}}`

Reviewer: ```suggestion
- Generate a PKGBUILD file:

`debtap --pkgbuild {{path/to/package.deb}}`
```
